backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTableManager.java

```diff
@@ -83,4 +87,16 @@ public void clearAllTables() {
         }
         return executor.execute(QueryBuilder.truncate(name));
     }
+
+    private InitialisationStatus buildStatus(long count) {
```

Comment:
How about having each table returning its InitialisationStatus & having a reduction implemented as part of `InitialisationStatus` ?